### PR TITLE
ddl: Add a check before building jobs

### PR DESCRIPTION
**Pull request by zimulala**
Add a check before building jobs in the operation of adding index, dropping index and dropping column.

### CODE REVIEW

File: ddl/column.go [hunk 1]
```diff
@@ -454,6 +448,19 @@ func (d *ddl) onModifyColumn(t *meta.Meta, job *model.Job) error {
 	return nil
 }
 
+// We don't support dropping column with index covered now.
```
Comment (hanfei1991) [inline ddl/column.go]: These comment don't explain this function. Move those to line 226

File: ddl/ddl.go [hunk 2]
```diff
@@ -1054,8 +1052,13 @@ func (d *ddl) DropColumn(ctx context.Context, ti ast.Ident, colName model.CIStr)
 		return ErrCantDropFieldOrKey.Gen("column %s doesn't exist", colName)
 	}
 
+	tblInfo := t.Meta()
```
Comment (hanfei1991) [inline ddl/ddl.go]: move this above line 1060

File: ddl/ddl.go [hunk 3]
```diff
@@ -1368,12 +1376,15 @@ func (d *ddl) DropIndex(ctx context.Context, ti ast.Ident, indexName model.CIStr
 	if !ok {
 		return errors.Trace(infoschema.ErrDatabaseNotExists)
 	}
-
 	t, err := is.TableByName(ti.Schema, ti.Name)
 	if err != nil {
 		return errors.Trace(infoschema.ErrTableNotExists)
 	}
 
+	if _, indexInfo := isIndexPublic(indexName.L, t.Meta().Indices); indexInfo == nil {
```
Comment (hanfei1991) [inline ddl/ddl.go]: Use indexInfo or idx ? Keep it same as line 1278

File: ddl/ddl.go (hunk 2 above)
Comment (zimulala) [inline ddl/ddl.go]: Why?

**Comment by zimulala** [comment]
PTAL @disksing @hanfei1991

**Comment by hanfei1991** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**